tests/rotate/test-rotate-end-to-end.sh

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env bash
+set -euo pipefail
+# test-rotate-end-to-end.sh
+# Runs end to end tests for the rotate module of spokewrench
+# Copyright Jackson M. Tsuji, 2024
+
+# If input field is not the appropriate length, print help and end script
+if [ $# != 5 ]; then
+  echo "$(basename $0): Run end to end test for the rotate module of spokewrench"
+  echo ""
+  echo "Usage: $(basename $0) input.fasta guide-positions.tsv guide-fractions.tsv expected_dir output_dir"
+  echo ""
+  echo "Positional arguments:"
+  echo "   input.fasta: input FastA file for test sequence rotation"
+  echo "   guide-positions.tsv: position-based guide file for test rotation"
+  echo "   guide-fractions.tsv: fraction-based guide file for test rotation"
+  echo "   expected_dir: directory containing expected output files from testing"
+  echo "   output_dir: directory where output files will be saved"
+  echo ""
+  exit 0
+fi
+
+# Set input variables
+input_fasta="$1"
+guide_positions="$2"
+guide_fractions="$3"
+expected_dir="$4"
+output_dir="$5"
+
+# Test input variables
+if [[ ! -f "${input_fasta}" ]]; then
+  echo "ERROR: Input file '${input_fasta}' does not exist."
+  exit 1
+elif [[ ! -f "${guide_positions}" ]]; then
+  echo "ERROR: Input file '${guide_positions}' does not exist."
+  exit 1
+elif [[ ! -f "${guide_fractions}" ]]; then
+  echo "ERROR: Input file '${guide_fractions}' does not exist."
+  exit 1
+elif [[ ! -d "${expected_dir}" ]]; then
+  echo "ERROR: Input directory '${expected_dir}' does not exist."
+  exit 1
+elif [[ -d "${output_dir}" ]]; then
+  echo "ERROR: Output directory '${output_dir}' already exists."
+  exit 1
+fi
+
+echo "Running tests..."
+mkdir -p "${output_dir}"
+
+# Do not exit early if errors are produced during testing
+set +e
+
+# Test series 1: midpoint rotation:
+# Test 1a: midpoint rotate, keeping sequence comment lines (should finish)
+spokewrench rotate -i "${input_fasta}" -o "${output_dir}/test1a.fasta" -r "${output_dir}/test1a.tsv" -v -m \
```

Comment:
Possible enhancement: to simplify the next several lines of code, a function could be created that specifies the skeleton syntax for the `rotate` command. Then, just the specific flags used for each test could be provided as input for the function, along with the base name for the output files.